File: src/main/java/com/study/platform/domain/user/model/UserRepository.java
package com.study.platform.domain.user.model;

import java.util.Collection;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface UserRepository {

    User save(User user);
    Optional<User> findById(UUID id);
    Optional<User> findByKakaoId(String kakaoId);
    Optional<User> findByNickname(String nickname);
    List<User> findAllByNicknameIn(Collection<String> nicknames);
}

File: domain/src/main/kotlin/com/study/platform/domain/user/model/UserRepository.kt
package com.study.platform.domain.user.model

import java.util.UUID

interface UserRepository {

    fun save(user: User): User
    fun findById(id: UUID): User?
    fun findByKakaoId(kakaoId: String): User?
    fun findByNickname(nickname: String): User?
    fun findAllByNicknameIn(nicknames: Collection<String>): List<User>
}
